src/reflection.jl

```diff
@@ -58,9 +58,13 @@ function code_typed(@nospecialize(job::CompilerJob); interactive::Bool=false, kw
         descend_code_typed(job.source.f, job.source.tt; interp, kwargs...)
     elseif VERSION >= v"1.7-"
         interp = get_interpreter(job)
-        InteractiveUtils.code_typed(job.source.f, job.source.tt; interp, kwargs...)
+        u = Base.unwrap_unionall(job.source.tt)
+        tt = Base.rewrap_unionall(Tuple{job.source.f, u.parameters...}, job.source.tt)
```

Comment:
Is this not just `Base.signature_type(job.source.f, job.source.tt)` and if not we should factor this out.

Reply:
It is not, base.signature_type has an ambiguity if you have a closure in that it does a check to see if it is a value or a type, and depending on which calls an additional TypeOf. The problem with the closure is that it will perform the wrong check if we know it is a type of a closure and instead get the closure constructor instead of calling the closure.

Reply:
Huh, do you have a concrete example?